src/main/java/de/tum/in/www1/artemis/service/metis/conversation/ConversationDTOService.java

```diff
@@ -0,0 +1,208 @@
+        this.channelAuthorizationService = channelAuthorizationService;
+    }
+
+    /**
+     * Creates a ConversationDTO from a Conversation
+     *
+     * @param conversation   the conversation to create the DTO from
+     * @param requestingUser the user requesting the DTO
+     * @return the created ConversationDTO
+     */
+    public ConversationDTO convertToDTO(Conversation conversation, User requestingUser) {
+        if (conversation instanceof Channel channel) {
+            return convertChannelToDto(requestingUser, channel);
+        }
+        if (conversation instanceof OneToOneChat oneToOneChat) {
+            return convertOneToOneChatToDto(requestingUser, oneToOneChat);
+        }
+        if (conversation instanceof GroupChat groupChat) {
+            return convertGroupChatToDto(requestingUser, groupChat);
+        }
+        throw new IllegalArgumentException("Conversation type not supported");
+    }
+
+    /**
+     * Creates a ConversationDTO from a Conversation
+     * <p>
+     * Does not set transient properties that require extra database queries
+     *
+     * @param conversation the conversation to create the DTO from
+     * @return the created ConversationDTO
+     */
+    public ConversationDTO convertToDTOWithNoExtraDBCalls(Conversation conversation) {
+        if (conversation instanceof Channel channel) {
+            return new ChannelDTO(channel);
+        }
+        if (conversation instanceof OneToOneChat oneToOneChat) {
+            return new OneToOneChatDTO(oneToOneChat);
+        }
+        if (conversation instanceof GroupChat groupChat) {
+            return new GroupChatDTO(groupChat);
+        }
+        throw new IllegalArgumentException("Conversation type not supported");
+    }
+
+    /**
+     * Creates a ChannelDTO from a Channel
+     *
+     * @param requestingUser the user requesting the DTO
+     * @param channel        the channel to create the DTO from
+     * @return the created ChannelDTO
+     */
+    @NotNull
+    public ChannelDTO convertChannelToDto(User requestingUser, Channel channel) {
+        var channelDTO = new ChannelDTO(channel);
+        channelDTO.setIsChannelModerator(channelAuthorizationService.isChannelModerator(channel.getId(), requestingUser.getId()));
+        channelDTO.setHasChannelModerationRights(channelAuthorizationService.hasChannelModerationRights(channel.getId(), requestingUser));
+        var participantOptional = conversationParticipantRepository.findConversationParticipantByConversationIdAndUserId(channel.getId(), requestingUser.getId());
+        channelDTO.setIsMember(participantOptional.isPresent());
+        participantOptional.ifPresent(conversationParticipant -> channelDTO.setLastReadDate(conversationParticipant.getLastRead()));
+        participantOptional.ifPresent(conversationParticipant -> channelDTO.setUnreadMessagesCount(conversationParticipant.getUnreadMessagesCount()));
```

Comment:
You can optimize this by merging the two lines.

Reply:
Done in 11008eb152db6fbddc6487737ac0b019b0b742b7